core/src/main/scala/org/apache/spark/util/ThreadUtils.scala

```diff
@@ -130,7 +224,15 @@ private[spark] object ThreadUtils {
    */
   def newDaemonCachedThreadPool(prefix: String): ThreadPoolExecutor = {
```

Comment:
Can you point out where we call these `ThreadUtil` methods to create threads pool at the executor side?

If your use case is for the driver side, can we document the usage? e.g. users can set MDC properties via `MDC.put` and they can see them in the driver logs, produced by `DAGScheduler` etc.

Reply:
without my code, it will not work hence the MDC will not passed to the thread pull.

Reply:
in my usecase I set the MDC in job-server so the driver will use it. but without the MDCAware it will not work. so I don't understand what you want to document.

Reply:
Does your job server use these `ThreadUtil` methods? If it does then please move these `MDCAware*` classes to your job server codebase.

The executor logs can print the `taskName` via MDC without all these changes in `ThreadUtil`. The use case you documented in https://github.com/apache/spark/pull/26624/files#diff-76e731333fb756df3bff5ddb3b731c46R2958 does not need these changes.

Reply:
My code just set the MDC and the only way to pass MDC between ThreadPools is the code I add to the ThreadUtils.
the executor creates threads pool, so in order that all logs of the executor will have the MDC populated you need that change.

Reply:
Let's make sure we are on the same page. What we document is:
1. users can set MDC properties via `spark.sparkContext.setLocalProperty("mdc." + name, "value")`
2. Spark framework will send the local properties to the executor side and set as task properties
3. We get the MDC properties from the task properties and apply them via `MDC.put`, in `Executor.run`

We don't need to propagate MDC properties through thread pools at all.

Reply:
without propagating it, almost all logs will not have the MDC. so that worse nothing.

Reply:
What is the propagating you are talking about? A common Spark application looks like
```
val session = ...
session.sql(...)
// other jobs
session.stop()
```
If you are using a job server and your job server submits Spark jobs in a thread pool, please propagate the MDC properties in that thread pool. But that thread pool is not part of Spark.

If you are referencing to other thread pools inside Spark, please point it out.

Reply:
AFAIK, the thread pool is part of spark - unless we have different views of what "spark" is.
MDC properties must pass through to the thread pool that is created by spark - hence it is part of spark.
Am I missing anything here?

Reply:
Please note that, `private[spark] object ThreadUtils` means it's private, and is supposed to be only used within Spark codebase. AFAIK the usages of `ThreadUtils` inside Spark don't need to propagate the MDC properties.

If you use these private APIs in your Spark application, and has new requirements, please fix it in your Spark application. You can simply put these `MDCAware*` classes in your application.

Reply:
ok, I will move the ThreadUtil change to another pool request, will need to set MDC in `TaskReaper` also.

Reply:
@cloud-fan pushed changes. please review